__unported__/base_delivery_carrier_files/carrier_file.py

```diff
@@ -147,15 +130,29 @@ def generate_files(self, cr, uid, carrier_file_id, picking_ids,
                                 'only one carrier at a time.')
             else:
                 carrier_file_id = carrier_file_id[0]
-        carrier_file = self.browse(cr, uid, carrier_file_id, context=context)
-        return self._generate_files(cr, uid, carrier_file, picking_ids,
-                                    context=context)
-
-
-class delivery_carrier(orm.Model):
+        carrier_file = self.browse(carrier_file_id)
+        return self._generate_files(carrier_file, picking_ids)
```

Comment:
Here you can call

    carrier_file._generate_files(picking_ids)

And rewrite _generate_files to use self instead of carrier_file